src/python/pants/backend/javascript/goals/lockfile_test.py

```diff
@@ -64,14 +64,20 @@ def given_package_with_name(name: str) -> str:
     return json.dumps({"name": name, "version": "0.0.1"})
 
 
-def given_package_with_workspaces(
-    name: str, version: str, dependencies: dict[str, str] | None = None, *workspaces: str
+def given_package_with_package_manager_and_workspaces(
```

Comment:
It might be a good idea to pin the package managers explicitly in these tests. But looking at the corepack version you're enabling here, the versions it proposes are:
```json
{
  "npm": "10.8.1",
  "pnpm": "9.2.0",
  "yarn": "4.2.2"
}
```

Reply:
I didn't attempt Yarn 1 -> Yarn 4 yet; `pnpm` version 9 definitely broke some stuff as I mentioned in the OP, so I pinned it to 7 instead. I'm going to attempt changing that to pnpm 8, and then change yarn 1 to yarn 4 and see what happens.